### PR TITLE
code cleanup: String to StringBuilder wherever applicable

### PR DESCRIPTION
**Pull request by preveen-stack**
leverage StringBuilder optimisation cleanup primitives in-built into the eclipse

### CODE REVIEW

File: ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/TargetDefinitionManager.java [hunk 1]
```diff
@@ -78,10 +78,10 @@ public int compare(IConfigurationElement o1, IConfigurationElement o2) {
 			}
 
 			private String getString(IConfigurationElement elem) {
-				String name = elem.getAttribute("name"); //$NON-NLS-1$
+				StringBuilder name = new StringBuilder().append(elem.getAttribute("name")); //$NON-NLS-1$
 				String id = elem.getAttribute("id"); //$NON-NLS-1$
-				name = name + " [" + id + "]"; //$NON-NLS-1$ //$NON-NLS-2$
-				return name;
+				name.append(" [").append(id).append("]"); //$NON-NLS-1$ //$NON-NLS-2$
+				return name.toString();
```
Comment (HannesWell) [inline ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/TargetDefinitionManager.java:84]:
```suggestion
				return = name + " [" + id + "]"; //$NON-NLS-1$ //$NON-NLS-2$
```

File: ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/TargetDefinitionManager.java [hunk 2]
```diff
@@ -78,10 +78,10 @@ public int compare(IConfigurationElement o1, IConfigurationElement o2) {
 			}
 
 			private String getString(IConfigurationElement elem) {
-				String name = elem.getAttribute("name"); //$NON-NLS-1$
+				StringBuilder name = new StringBuilder().append(elem.getAttribute("name")); //$NON-NLS-1$
```
Comment (HannesWell) [inline ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/TargetDefinitionManager.java:81]: A StringBuilder is not necessary for static string concatenation. The Java compiler does even smarter things under the hood any ways.
```suggestion
				String name = elem.getAttribute("name"); //$NON-NLS-1$
```

**Review by HannesWell — CHANGES_REQUESTED** [review]
Thank you for the contribution.
But not in all cases it makes sense to use a `StringBuilder`.

**Comment by laeubi** [comment]
@preveen-stack can you consider the suggested changes and rebase the branch?

**Comment by preveen-stack** [comment]
Going through some personal difficulties at the moment. Closing this for now. Will reattempt as soon as possible. The inconvenience caused is deeply regretted. Thanks for understanding.

**Pull request closed without merging**